### DISCUSSION TITLE
What's the best way to approach a parametrized input transformation?

### DISCUSSION THREAD

**Discussion by Aceticia** [manujosephv/pytorch_tabular#447] [Q&A]
Hi all, I have a specific use case: For my work, I have tables with spatial coordinates. I found from a preliminary model that processing the spatial coordinates by learning a MLP with fourier feature creates a good representation, and I'm wondering what's the best way I can achieve this in `pytorch_tabular`. So in summary, learning a custom MLP as a preprocessing step for a subset of the columns. I can write a whole new custom model, but I'd also like to take advantage of all the other features provided by `pytorch_tabular`.

It's possible to just add a simple Fourier feature layer as an input transformation, but sadly learning a single vector to represent the 3 spatial location before they interact with other columns is crucial for the task I'm working on since downstream tasks rely on accessing these spatial embeddings later.

And if you would like to hear more, I'd be happy to describe more details.

**Comment by manujosephv** [comment]
Interesting.. You can check the documentation/tutorials. There is a tutorial on how to write custom models and make it work with Pytorch Tabular
